components/sup/src/manager/service/spec.rs

```diff
@@ -337,27 +337,94 @@ impl ServiceSpec {
+                    // way, however. Initially we are allowing dynamic
+                    // update of update-related configuration, but
+                    // nothing prevents us from making more things
+                    // dynamic in the future. We are proceeding
+                    // conservatively.
+
+                    // NOTE: There is no need to compare desired
+                    // state, because we know they must both be `Up`
+                    // for us to be in this code.
+
+                    // NOTE: if the idents change in any way, you
+                    // *must* restart, since that change may result in
+                    // a different version of the service being run.
+                    if running_spec.ident != disk_spec.ident
+                        || running_spec.group != disk_spec.group
+                        // TODO (CM): This *might* not need to be here
+                        || running_spec.topology != disk_spec.topology
+                        // TODO (CM): Bind information *may* be able
+                        // to be dynamically changed, but that will
+                        // need to be investigated more deeply.
+                        || running_spec.binds != disk_spec.binds
+                        || running_spec.binding_mode != disk_spec.binding_mode
+                        || running_spec.config_from != disk_spec.config_from
+                        // TODO (CM): This probably doesn't need to be here
+                        || running_spec.shutdown_timeout != disk_spec.shutdown_timeout
+                        || running_spec.svc_encrypted_password != disk_spec.svc_encrypted_password
+                        // TODO (CM): This probably doesn't need to be here, either
+                        || running_spec.health_check_interval != disk_spec.health_check_interval
+                    {
+                        debug!("Reconciliation: '{}' queued for restart",
+                               running_spec.ident);
+                        Some(ServiceOperation::Restart { to_stop:  running_spec,
+                                                         to_start: disk_spec, })
+                    } else {
+                        let mut ops = HashSet::new();
+                        if running_spec.bldr_url != disk_spec.bldr_url
+                            || running_spec.channel != disk_spec.channel
+                            || running_spec.update_strategy != disk_spec.update_strategy
+                            || running_spec.update_condition != disk_spec.update_condition
+                        {
+                            ops.insert(RefreshOperation::RestartUpdater);
+                        }
+
+                        // We should have *something* to do down
+                        // here, but if we don't, let's be explicit
+                        // about it.
+                        if ops.is_empty() {
+                            None
```

Comment:
Would it be worth adding a `debug_assert` and error log here? If we ever reach this it seems like something has to be wrong with our logic.